c_api/CMakeLists.txt

```diff
@@ -0,0 +1,21 @@
+# Copyright (c) Facebook, Inc. and its affiliates.
+#
+# This source code is licensed under the MIT license found in the
+# LICENSE file in the root directory of this source tree.
+
+cmake_minimum_required(VERSION 3.17 FATAL_ERROR)
+
+project(faiss_c_library LANGUAGES C CXX)
+
+set(CMAKE_C_STANDARD 11)
+
+file(GLOB SOURCES "${CMAKE_SOURCE_DIR}/c_api/*.cpp")
+add_library(faiss_c STATIC ${SOURCES})
```

Comment:
Please add `EXCLUDE_FROM_ALL` to the library definition so that it is not build by default, but only when explicitly building the `faiss_c` target (cf. https://cmake.org/cmake/help/latest/command/add_library.html#normal-libraries).